docs/index.md

```diff
@@ -3984,6 +3986,116 @@ Here are some common symbols defined in `set.mm`.
   the <tt>tan(A)</tt> notation used by others
   but without context-dependent notational ambiguity.
 
+### JSON format
+
+Metamath-lamp exports proofs in JSON format, which you can later
+import to restore the current state.
+We recommend using the `.lamp.json` extension if you store them
+in a file.
+
+#### JSON format specification
+
+The JSON format has a single object at its root, with the
+following keys and values:
+
+* "srcs": Array of source objects (see below).
+  This array collectively defines the proof context.
+* "descr": String. The description.
+* "varsText": String. The text of the variables.
+* "disjText": String. The text of the disjunctions.
+* "stmts": Array of step objects (see below).
+
+A source object has the following keys and values:
+
+* "typ": String. It should be "Web" (load from the web) or "Local".
+* "fileName": String. The filename (if "typ" is "Local").
+* "url": String. The URL (if "typ" is "Web").
+* "readInstr": String. This can be "StopBefore", "StopAfter", or "ReadAll".
+* "label": String. This is the label to "StopBefore" or "StopAfter".
+* "resetNestingLevel": Boolean.
+* "allLabels": Array.
```

Comment:
"allLabels" is not required when importing from json. If you remove it, the import will still work.

Reply:
Thanks! I might be best to eventually remove it from the export, but it's clearly harmless so that's certainly not urgent. Thanks for the explanations.